### DISCUSSION TITLE
Does the S3 File Upload component work with other S3-compatible services or only from AWS?

### DISCUSSION THREAD

**Discussion by daawaan4x** [Budibase/budibase#5459] [Help]
Hello, I've been trying to use the S3 File Upload component since the 20MB file upload limit for attachments is too small for our use case which is explained further in #5457. Unfortunately, the S3 component isn't showing any of the S3 datasources in the dropdown for the component which I've already issued in #5454. We're using digitalocean spaces in our case and I was wondering if our datasource wasn't showing up because it isn't from AWS? Its also stated here in #4030 the first pull request for the S3 File Upload component that

> S3 datasources with custom endpoints are not supported. This is mainly due to the non-compliance with presigned URL domain names. Real S3 uses bucket.domain as the FQDN, and using a custom endpoint when generating a presigned URL spits out a value of something like bucket.10.0.0.1 for example, if using a local IP address. Due to this behaviour we just have to put a limitiation on this functionality that it will only work with real S3 and not MinIO.

so I think it implies that the S3 File Upload component only accepts S3 endpoints from AWS? I'd like to hear some clarification for this

**Comment by ybotmallah** [comment]
I am also facing the same issue as described here.

I think you are right in thinking that unless you are using an actual S3 endpoint rather than an S3 compatible endpoint, you can only see uploaded files.

It would be good to add support for any custom S3 endpoints. My chosen cloud provider is DigitalOcean, it would be nice to keep all of my services in one place instead of having to use AWS.

**Comment by melohagan** [comment]
Hey @daawaan @ybotmallah 

You are correct, currently the S3 Upload component only supports AWS.

I have created an issue to expand the support for this: https://github.com/Budibase/budibase/issues/5587
